paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/catalyst/analysis/PaimonIncompatiblePHRRules.scala

```diff
@@ -43,6 +43,11 @@ case class PaimonIncompatiblePHRRules(session: SparkSession) extends Rule[Logica
         }.toMap
         PaimonTruncateTableCommand(table, partitionSpec)
 
+      case t @ ShowPartitions(PaimonRelation(table), specOpt, _) if t.resolved =>
+        PaimonShowTablePartitionsCommand(
+          table,
+          specOpt.map(s => s.asInstanceOf[ResolvedPartitionSpec]))
```

Comment:
try this, maybe it can work too:
```
import org.apache.spark.sql.execution.datasources.v2.DataSourceV2Implicits._
specOpt.asResolvedPartitionSpecs
```

Reply:
> asResolvedPartitionSpecs

This is useful for `Seq[PartitionSpec]`.